scripts/webhook-migration/updateWebhooks.sh

```diff
@@ -0,0 +1,223 @@
+findGitHubRepos() {
+  export KUBECONFIG=/tmp/config
+  echo "Using KUBECONFIG ${KUBECONFIG}"
+  oc login ${OPENSHIFT_URL} --token=${OPENSHIFT_API_TOKEN} > /dev/null
+
+  local projects=$(oc projects -q)
+  for project in ${projects}; do
+   echo "Processing project ${project}"
+   oc project ${project} > /dev/null;
+   local tmp=($(oc get bc -o json | jq -r ' .items[] | select(.spec.source.git.uri) | .metadata.namespace + " " + .spec.source.git.uri[19:-4]'))
+   if [ ${#tmp[@]} -eq 0  ]; then
+    continue
+   fi
+   NAME_SPACES+=(${tmp[0]})
+   REPOS+=(${tmp[1]})
+  done
+}
+
+###############################################################################
+# Finds the UUIDs for the discovered namespaces
+# Globals:
+#   NAME_SPACES - Namespaces to iterate
+#   UUIDS       - Array which gets populated with UUIDs
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+findUUIDs() {
+  for i in "${!NAME_SPACES[@]}"; do
+    uuid=$(curl -sgSL -H "Authorization: Bearer ${AUTH_TOKEN}" ${TENANT_URL}api/tenants?namespace=${NAME_SPACES[$i]}\&master_url=${OPENSHIFT_URL} | jq -r .data[0].id)
+     if [ ${#uuid} -le 5 ]; then
+        echo "WARN: Unable to determine UUID for ${NAME_SPACES[$i]}." >&2
+        echo "WARN: Request URL ${TENANT_URL}api/tenants?namespace=${NAME_SPACES[$i]}\&master_url=${OPENSHIFT_URL}" >&2
+      else
+        UUIDS[$i]=${uuid}
+      fi
+  done
+}
+
+###############################################################################
+# For each UUID get an OSIO token
+# Globals:
+#   UUIDS      - User ids to iterate
+#   OSIO_TOKEN - Array which gets populated with OSIO tokens
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+getOsioTokens() {
+  for i in "${!UUIDS[@]}"; do
+    osio_token=$(osio -t ${TARGET_ENV} token -k "${PRIVATE_KEY}" -u ${UUIDS[$i]})
+    OSIO_TOKENS[$i]="${osio_token}"
+  done
+}
+
+###############################################################################
+# For each OSIO token get the GitHub token
+# Globals:
+#   OSIO_TOKENS    - OSIO tokens to iterate
+#   GITHUB_TOKENS  - Array which gets populated with GitHub tokens
+#   UUIDS          - User ids (for logging)
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+getGitHubTokens() {
+  for i in "${!OSIO_TOKENS[@]}"; do
+    [ -z "${OSIO_TOKENS[${i}]}"  ] && echo "WARN: Empty OSIO token for namespace ${NAME_SPACES[$i]}" >&2 && continue
+
+    gh_token=$(curl -sgSL -H "Authorization: Bearer ${OSIO_TOKENS[${i}]}" ${AUTH_URL}api/token?for=https://github.com | jq -r '.access_token')
+    if [ "${gh_token}" == "null" ]; then
+        echo "WARN: Unable to create GitHub token for UUID ${UUIDS[$i]}" >&2
+        curl -sgSL -H "Authorization: Bearer ${OSIO_TOKENS[${i}]}" ${AUTH_URL}api/token?for=https://github.com >&2
```

Comment:
Is this just to get output for the curl command to STDERR?

Reply:
Yes, I found this helpful for the cases where something failed.